tensorboard/webapp/types/data.ts

```diff
@@ -22,8 +22,17 @@ export enum DataLoadState {
   FAILED,
 }
 
+export enum LoadFailureCode {
+  UNKNOWN = 'UNKNOWN',
+  NOT_FOUND = 'NOT_FOUND',
+}
+
 export interface LoadState {
   state: DataLoadState;
-  // Time since epoch.
+  // Time of last successful load. Time since epoch.
   lastLoadedTimeInMs: number | null;
+  // Reason for failure of most recently completed request. This should not be
+  // set if there has not been a failure or if the most recently completed
```

Comment:
> This should not be set if there has not been a failure

Can you guarantee this with typing instead of comment?

Reply:
I can guarantee for state === NOT_LOADED, LOADED, FAILED but there is still some ambiguity for state === LOADING.

I also made slight adjustments for typing of lastLoadedTimeInMs.

I feel like I'll have a bad time trying to get this into the internal repo - will start looking at that now.